modules/textures/src/lib/parsers/parse-npy.ts

```diff
@@ -48,6 +59,13 @@ export function parseNPY(arrayBuffer: ArrayBuffer, options?: unknown) {
   const nArrayElements = header.shape?.reduce((a: number, b: number): number => a * b);
   const arrayByteLength = nArrayElements * ArrayType.BYTES_PER_ELEMENT;
 
+  if (arrayBuffer.byteLength < headerEndOffset + arrayByteLength) {
+    // eslint-disable-next-line no-console, no-undef
+    console.warn(
+      `Buffer not long enough for slice: ${headerEndOffset}-${headerEndOffset + arrayByteLength}`
+    );
+    return null;
```

Comment:
Is this better than throwing?

Reply:
Previously we logged a warning and returned null, but we can change that